src/providers/oneNoteApiDataProvider.ts

```diff
@@ -151,12 +151,13 @@ export class OneNoteApiDataProvider implements OneNoteDataProvider {
 		return decodeURIComponent(last);
 	}
 
-	getSpNotebookProperties(spNotebook: SharedNotebook, expands?: number, excludeReadOnlyNotebooks?: boolean): Promise<SharedNotebookApiProperties | undefined> {
+	getSpNotebookProperties(spNotebook: SharedNotebook, expands?: number, excludeReadOnlyNotebooks?: boolean): Promise<SharedNotebookApiProperties> {
 		return new Promise<SharedNotebookApiProperties>((resolve, reject) => {
 			this.getNotebookSelfUrlFromSpUrl(spNotebook.webUrl).then((selfUrl) => {
 				this.http('GET', selfUrl + '?' + this.getExpands(expands), this.authHeader, this.headers).then((xhr) => {
 					const notebook: OneNoteApi.Notebook = xhr.response && JSON.parse(xhr.response);
 					if (notebook) {
+						spNotebook.apiUrl = notebook.self;
```

Comment:
I never understood why we would map the original structure of the notebook that comes from the API (e.g. OneNoteApi.Notebook) to another data structure... it may be worth considering just using those types.